.github/workflows/build.yml

```diff
@@ -30,3 +28,28 @@ jobs:
           npm run init
           npm run git-info
           npm run build
+
+  test-firebase:
```

Comment:
Can we break this one down into 3 separate jobs, ie. build, deploy to test infrastructure, and run tests? It looks like GitHub is [dragging its feet](https://github.community/t5/GitHub-Actions/Ability-to-rerun-just-a-single-job-in-a-workflow/td-p/41629) on the obvious feature of being able to re-run specific jobs, but this will prepare us for the (possibly mythical) day they release that feature (for example, if our build succeeds but our test run fails, sometimes we just want to rerun the tests again without waiting for a re-build). Breaking this down will also prepare us for a more complex testing workflow that includes frontend tests and integration tests.

See [here](https://help.github.com/en/actions/configuring-and-managing-workflows/persisting-workflow-data-using-artifacts#passing-data-between-jobs-in-a-workflow) for how to persist data between jobs, and denote that certain jobs are dependent on other jobs.